pkg/testing/ess/serverless_provision.go

```diff
@@ -71,114 +62,82 @@ func (prov *ServerlessProvision) SetLogger(l runner.Logger) {
 	prov.log = l
 }
 
-// Provision a new set of serverless instances
-func (prov *ServerlessProvision) Provision(ctx context.Context, requests []runner.StackRequest) ([]runner.Stack, error) {
-	upWaiter := sync.WaitGroup{}
-	depErrs := make(chan error, len(requests))
-	depUp := make(chan bool, len(requests))
-	stacks := []runner.Stack{}
-	for _, req := range requests {
-		client := NewServerlessClient(prov.cfg.Region, "observability", prov.cfg.APIKey, prov.log)
-		srvReq := ServerlessRequest{Name: req.ID, RegionID: prov.cfg.Region}
-		proj, err := client.DeployStack(ctx, srvReq)
-		if err != nil {
-			return nil, fmt.Errorf("error deploying stack for request %s: %w", req.ID, err)
-		}
-		err = client.WaitForEndpoints(ctx)
-		if err != nil {
-			return nil, fmt.Errorf("error waiting for endpoints to become available for request: %w", err)
-		}
-		newStack := runner.Stack{
-			ID:            req.ID,
-			Version:       req.Version,
-			Elasticsearch: client.proj.Endpoints.Elasticsearch,
-			Kibana:        client.proj.Endpoints.Kibana,
-			Username:      client.proj.Credentials.Username,
-			Password:      client.proj.Credentials.Password,
-			Internal: map[string]interface{}{
-				"deployment_id":   proj.ID,
-				"deployment_type": proj.Type,
-			},
-		}
-		stacks = append(stacks, newStack)
-		prov.stacksMut.Lock()
-		prov.stacks[req.ID] = stackhandlerData{client: client, stackData: newStack}
-		prov.stacksMut.Unlock()
-
-		upWaiter.Add(1)
-		go func() {
-			isUp, err := client.DeploymentIsReady(ctx)
-			if err != nil {
-				depErrs <- err
-
-			}
-			depUp <- isUp
-		}()
-	}
-
-	gotUp := 0
-	for {
-		select {
-		case <-ctx.Done():
-			return nil, ctx.Err()
-		case err := <-depErrs:
-			return nil, fmt.Errorf("error waiting for stacks to become available: %w", err)
-		case isUp := <-depUp:
-			if isUp {
-				gotUp++
-			}
-			if gotUp >= len(requests) {
-				return stacks, nil
-			}
-		}
+// Create creates a stack.
+func (prov *ServerlessProvision) Create(ctx context.Context, request runner.StackRequest) (runner.Stack, error) {
+	// allow up to 4 minutes for requests
+	createCtx, createCancel := context.WithTimeout(ctx, 4*time.Minute)
+	defer createCancel()
+
+	client := NewServerlessClient(prov.cfg.Region, "observability", prov.cfg.APIKey, prov.log)
+	srvReq := ServerlessRequest{Name: request.ID, RegionID: prov.cfg.Region}
+
+	prov.log.Logf("Creating serverless stack %s (%s)", request.Version, request.ID)
+	proj, err := client.DeployStack(createCtx, srvReq)
+	if err != nil {
+		return runner.Stack{}, fmt.Errorf("error deploying stack for request %s: %w", request.ID, err)
```

Comment:
I am not sure if it's possible, at least from what I saw of the serverless API but it would be nice if we could tag the serverless projects like we do the ess deployments...

Reply:
I have no idea either.

Reply:
I don't recall seeing a tagging API when I was first writing for the serverless API, but that might have changed. In general the API is much more restrictive than the stateful API.